src/explorer.api/Models/DiffixApiClient.cs

```diff
@@ -254,16 +298,20 @@ async public Task<QueryResult<RowType>> PollQueryResult<RowType>(string queryId)
             throw new Exception("Should never reach here.");
         }
 
+        /// <summary>
+        /// Polls for a query's results until query resolution is complete, or until a specified timeout.
+        /// </summary>
+        /// <param name="queryId">The query Id obtained via a previous call to the /api/query endpoint.</param>
+        /// <param name="timeout">How long to wait for the query to complete.</param>
```

Comment:
Defaults to infinity? That's useful to know if so

Reply:
I think in the end I decided to make the timeout parameter a required parameter...